### DISCUSSION TITLE
Can the DataList access the "current node" for the use of the custom data source?

### DISCUSSION THREAD

**Discussion by hfloyd** [leekelleher/umbraco-contentment#123] [Question] [answered]
So, I was wondering if there was any way to get information about the **current node** the editor is being used-on, like how the Multi-node treepicker "knows" what node is in use - and then can use XPath to determine the "dataset" of available nodes to select from:
![image](https://user-images.githubusercontent.com/1252593/124678858-602b3280-de91-11eb-8d5f-7c4faded6c9c.png)

Just some way to access an INT id or UDI for using in the "GetItems()" function?

**Comment by leekelleher** [comment, marked as answer]
Hi @hfloyd.

It is possible to access the querystring parameters from custom data-source code. (See my reply on https://github.com/leekelleher/umbraco-contentment/issues/76#issuecomment-795753300.)

This is how I do it on the Umbraco Content data-source:
https://github.com/leekelleher/umbraco-contentment/blob/2.1.0/src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/UmbracoContentDataListSource.cs#L65

Hope this helps?

**Reply by hfloyd** [reply]
Thanks, @leekelleher, that gave me a great start. I was able to get the data I was looking for when the node loads... but when I choose a value and "Save & Publish" the node, GetItems() is running again - but the URL is "/umbraco/backoffice/UmbracoApi/Content/PostSave" - so the ID value isn't present.

I worked around it like this:

```
 public IEnumerable<DataListItem> GetItems(Dictionary<string, object> config)
  {
      var items = new List<DataListItem>();
      var fileName = config["ConfigFileName"].ToString();

      var currentNode = GetCurrentNodeId();

      //If no node Id - skip all this
     if (currentNode==0)
      {
          return Enumerable.Empty<DataListItem>();
      }

     items = ... Do other things with currentNode to get values list ...
      
      return items;
  }

 private int GetCurrentNodeId()
  {
      var nodeContextId = default(int?);
      var umbracoContext = _umbracoContextAccessor.UmbracoContext;

      // NOTE: First we check for "id" (if on a content page), then "parentId" (if editing an element).
      if (int.TryParse(umbracoContext.HttpContext.Request.QueryString.Get("id"), out var currentId) == true)
      {
          nodeContextId = currentId;
      }
      else if (int.TryParse(umbracoContext.HttpContext.Request.QueryString.Get("parentId"), out var parentId) == true)
      {
          nodeContextId = parentId;
      }

      //If no node found, it's not available
      if (nodeContextId == null) { return 0; }
      else
      {
          return (int) nodeContextId;
      }
  }

```
